brayns/common/scene/Model.h

```diff
@@ -186,16 +186,14 @@ class Model
         Returns the bounds for the Model
     */
     const Boxf& getBounds() const { return _bounds; }
-    template <typename T>
-    void updateBounds(const T& value)
-    {
-        _bounds.merge(value);
-    }
-    void setBounds(const Boxf& box) { _bounds = box; }
     /**
         Returns spheres handled by the Model
     */
-    BRAYNS_API SpheresMap& getSpheres() { return _spheres; }
+    BRAYNS_API SpheresMap& getSpheres()
+    {
+        _spheresDirty = true;
+        return _spheres;
+    }
```

Comment:
This looks really bizarre. You're assuming that the sphere map is going to be changed just be getting it?
At least you should add a const method that doesn't do that and returns a const SphereMap& and document the side effect of calling the non const method. Anyway, if you find a way to move the dirty flag to the SpheresMap, that's even better (I'm suggesting this without knowing the type of SpheresMap)

Reply:
I don't want to expose the spheres etc. like this at all like that, but that's the way it is designed by the original author. I just corrected that a non-const access potential modifies the spheres. A const version is not needed atm, but I can add it for sure.